src/lib.rs

```diff
@@ -884,6 +884,9 @@ impl Url {
 
     /// Return this URL’s fragment identifier, if any.
     ///
+    /// A fragment is the part of a url following the `#` symbol,
+    /// which is used to identify a location in the document.
```

Comment:
I don't think this is necessarily true of every URL fragment, especially if the URL does not refer to an HTML document. Could you phrase this in a more precise way? Check out [this article](https://en.wikipedia.org/wiki/URL#Syntax) for inspiration.